prow/bugzilla/client.go

```diff
@@ -75,25 +78,77 @@ type Client interface {
 	RemovePullRequestAsExternalBug(id int, org, repo string, num int) (bool, error)
 	// GetAllClones returns all the clones of the bug including itself
 	// Differs from GetClones as GetClones only gets the child clones which are one level lower
-	GetAllClones(bug *Bug) ([]*Bug, error)
+	// GetAllClones will also fetch bugDetails and the root bug details in addition to clones
+	// since we want to parallelize all the HTTP calls
+	GetAllClones(bugID int) (clones []*Bug, bug *Bug, root *Bug, err error)
 	// GetRootForClone returns the original bug.
 	GetRootForClone(bug *Bug) (*Bug, error)
+	// GetFromCache will fetch the clones from the clones cache
+	GetFromCache(key int) ([]int, bool)
+	// SetInCache will push the cloneIDs associated with the bug to the cache
+	SetInCache(key int, value []int)
 }
 
 func NewClient(getAPIKey func() []byte, endpoint string) Client {
 	return &client{
-		logger:    logrus.WithField("client", "bugzilla"),
-		client:    &http.Client{},
-		endpoint:  endpoint,
-		getAPIKey: getAPIKey,
+		logger:      logrus.WithField("client", "bugzilla"),
+		client:      &http.Client{},
+		endpoint:    endpoint,
+		getAPIKey:   getAPIKey,
+		clonesCache: clonesCacheWithMutex{cache: map[int][]int{}},
```

Comment:
The cache is not valid to use between calls, I thought this was supposed to be only for the double-traversal when we were running a `GetAllClones()` from the middle of a tree?

Reply:
I have 2 caches now. This one should be valid between calls and holds a list of clones associated with a bug to allow for parallelizing all the GetBug calls.
The other one is for the double traversal(ie. the bugCache) and is not used between calls

Reply:
I think the `clonesCache` is something for a dedicated PR and might make more sense downstream?